src/lib/comms/sol-coap.c

```diff
@@ -900,6 +910,8 @@ sol_coap_send_packet_with_reply(struct sol_coap_server *server, struct sol_coap_
     }
 
     sol_coap_header_get_id(pkt, &reply->id);
+    reply->reentrant.in_use = false;
+    reply->reentrant.delete_me = false;
```

Comment:
Actually, if we can assume that members are always initialized to 0, we don't need this.